src/tools/perf/lib/ucp_tests.cc

```diff
@@ -419,12 +419,17 @@ class ucp_perf_test_runner {
             *(volatile psn_t*)ptr = sn;
         } else {
             request = ucp_put_nbx(m_perf.ucp.self_ep, &sn, sizeof(sn),
-                                  (uint64_t)ptr, m_perf.ucp.self_send_rkey,
-                                  &param);
+                                  (uint64_t)ptr, rkey, &param);
             request_wait(request, mem_type, "write_sn");
         }
     }
 
+    UCS_F_ALWAYS_INLINE void write_sn(void *buffer, ucs_memory_type_t mem_type,
+                                      size_t length, psn_t sn)
+    {
+        write_sn_ext(buffer, mem_type, length, sn, m_perf.ucp.self_send_rkey);
+    }
```

Comment:
why not always pass rkey to write_sn? IMO this wrapper is redundant..

Reply:
because in most cases it will be self_send_rkey, I'll remove the wrapper anyway